engine/consensus/matching/engine.go

```diff
@@ -98,1063 +75,169 @@ func New(
 	assigner module.ChunkAssigner,
 	validator module.ReceiptValidator,
 	requiredApprovalsForSealConstruction uint,
-	emergencySealingActive bool,
-) (*Engine, error) {
-
-	// initialize the propagation engine with its dependencies
-	e := &Engine{
+	emergencySealingActive bool) (*Engine, error) {
+	// create channels that will be used to feed data to matching core.
+	receiptsChannel := make(chan *Event)
+	approvalsChannel := make(chan *Event)
+	approvalResponsesChannel := make(chan *Event)
```

Comment:
would it be worth making these buffered channels?
Otherwise pushing items to them will block until the core is done processing the event.
Not sure if this is expected behaviour.

Reply:
While we can add some buffering, this object is designed to work in non blocking manner, it really depends on processPendingEvent. If it doesn't do anything heavy then we won't block anything.